engine/plugins/state_machine/action_handler.js

```diff
@@ -346,6 +346,95 @@ class ActionHandler {
             client.state.items[itemIndex] = client.actionData;
         }
     }
+
+    /**
+     * Get status for the user of the client.
+     *
+     * @param client
+     *   The client.
+     */
+    status(client) {
+        client.state.step = 'status';
+
+        const priorState = client.__machina__.bibbox.priorState;
+        const statusRefreshing = ['checkOutItems', 'checkInItems'].includes(priorState);
+
+        // Information is already present after the login in client.internal.
+        client.state = Object.assign({}, client.state, {
+            // Status refreshing. Determines if a a refresh of the status is under way.
+            statusRefreshing: statusRefreshing,
+            // Items that are ready to be picked up.
+            holdItems: client.internal.user.holdItems,
+            // Items that are overdue being checked in.
+            overdueItems: client.internal.user.overdueItems,
+            // Items the user has checked out.
+            chargedItems: client.internal.user.chargedItems,
+            // Items with a fine.
+            fineItems: client.internal.user.fineItems,
+            // Items that have been recalled.
+            recallItems: client.internal.user.recallItems,
+            // Items the user has reserved, but which are not ready.
+            unavailableHoldItems: client.internal.user.unavailableHoldItems
+        });
+
+        if (statusRefreshing) {
+            const busEvent = uniqid('fbs.patron.');
+            const errEvent = uniqid('fbs.patron.err.');
+
+            /**
+             * Listen for patron success event.
+             */
+            this.bus.once(busEvent, resp => {
+                debug('Patron retrieved successfully');
+                debug(resp);
```

Comment:
Could this send private data into debugger module? Which is GDPR related?

Reply:
Good point. I will remove the debug(resp);